assets/hmr.js

```diff
@@ -23,9 +25,13 @@ class HotModuleState {
     this.disposeCallbacks.push(callback);
   }
   accept(callback = true) {
-    if (this.isLocked) {
-      return;
+    if (!this.isAccepted) {
+      socket.addEventListener('open', () => {
```

Comment:
this feels like we're just asking for a race condition :) lets move this logic up into a `sendMessage()` method, where it either sends the message immediately (if connection is already open) or adds it to a queue to be sent when the connection finally does open (via `socket.addEventListener('open', ...`).